docs/feature_programmable_button.md

```diff
@@ -0,0 +1,68 @@
+## Programmable Button
+
+Programmable button is a feature that can be used to send keys that have no
```

Comment:
The real question is, how is this used host side?

Reply:
This strongly depends on the host OS.
On Linux (> 5.14) this will be handled by the default HID layer and emit `KEY_MACRO#` keycodes (Only up to `KEY_MACRO30` for know).
With other OSs I'm not really familiar enough to know.

Reply:
I added an explanation to the docs.